### PR TITLE
Communities menu: use flask menu to generate communities menu

### PR DESCRIPTION
**Pull request by Pineirin**
closes: inveniosoftware/invenio-communities#408

### CODE REVIEW

File: invenio_communities/views/ui.py [hunk 1]
```diff
@@ -113,6 +120,38 @@ def register_menus():
             order=3,
         )
 
+        communities = current_menu.submenu('communities')
+        communities.submenu('requests').register(
+            'invenio_communities.communities_requests',
+            text=_('Requests'),
+            order=2,
+            expected_args=["pid_value"],
+            **communities_kwargs(
```
Comment (zzacharo) [inline invenio_communities/views/ui.py:129]: Same comment with https://github.com/inveniosoftware/invenio-app-rdm/pull/1467#discussion_r857321158

File: invenio_communities/views/ui.py [hunk 2]
```diff
@@ -113,6 +120,38 @@ def register_menus():
...
+                'comments icon', 
+                'can_search_requests'
+            )
+        )
+        communities.submenu('members').register(
+            'invenio_communities.members',
+            text=_('Members'),
+            order=3,
+            expected_args=["pid_value"],
+            **communities_kwargs(
```
Comment (zzacharo) [inline invenio_communities/views/ui.py:139]: Same comment with https://github.com/inveniosoftware/invenio-app-rdm/pull/1467#discussion_r857321158

File: invenio_communities/views/ui.py [hunk 3]
```diff
@@ -113,6 +120,38 @@ def register_menus():
...
+                'users icon', 
+                'can_read' 
+            )
+        ) 
+        communities.submenu('settings').register(
+            'invenio_communities.communities_settings',
+            text=_('Settings'),
+            order=4,
+            expected_args=["pid_value"],
+            **communities_kwargs(
```
Comment (zzacharo) [inline invenio_communities/views/ui.py:149]: Same comment with https://github.com/inveniosoftware/invenio-app-rdm/pull/1467#discussion_r857321158

File: invenio_communities/templates/semantic-ui/invenio_communities/details/header.html [hunk 4]
```diff
@@ -79,15 +79,9 @@ <h2 class="ui header">{{ community.metadata.title }}</h2>
...
-    {% set header_menu_items = {
-      'search': (_('Search'), "/communities/" + community.id, 'search icon', True), 
-      'requests': (_('Requests'), url_for('invenio_communities.communities_requests', pid_value=community.id), 'comments icon', permissions.can_search_requests),
-      'members': (_('Members'), url_for('invenio_communities.members', pid_value=community.id), 'users icon', permissions.can_read),
-      'settings': (_('Settings'), url_for('invenio_communities.communities_settings', pid_value=community.id), 'settings icon', permissions.can_update),
-    } %}
     <div class="ui container secondary pointing menu page-subheader pl-0">
-      {% for header_menu_key, (header_menu_item, header_menu_link, menu_icon, permission_to_show ) in header_menu_items.items() if permission_to_show %}
-        <a class="item {{ 'active' if active_header_menu_item == header_menu_key }} {{ 'disabled' if not header_menu_link }}" href="{{ header_menu_link }}"><i aria-hidden="true" class="{{menu_icon}}"></i>{{ header_menu_item }}</a>
+      {% for item in current_menu.submenu('communities').children if item.permissions == True or permissions[item.permissions]  %}
```
Comment (kpsherva) [inline invenio_communities/templates/semantic-ui/invenio_communities/details/header.html:83]:
```suggestion
      {% for item in current_menu.submenu('communities').children if item.permissions or permissions[item.permissions]  %}
```
why exactly do we need `permissions[item.permissions]` in the conditional statement ? what does it do?

Comment (kpsherva) [reply invenio_communities/templates/semantic-ui/invenio_communities/details/header.html:83]: discussed directly

**Review by kpsherva — APPROVED** [review]
for housekeeping: we left the icon as a parameter for now until better solution found

the discussion was on the validity of keeping frontend attributes (icon name) in the backend settings (flask-menu). Flask menu implementation enforces us for now to place the icon in the menu items registration, until better solution is found

**Review by kpsherva — APPROVED** [review]

**Pull request merged**